### DISCUSSION TITLE
Default hostname when providing a nullish `host` in listening options?

### DISCUSSION THREAD

**Discussion by driimus** [fastify/fastify#5582] [Ideas] [answered]
## Context
Fastify currently listens on `localhost` by default, but there are still scenarios where that default isn't propagated to `listenOptions`.

One example of this is when listening options are provided and the `host` property is falsy:
```js
 require('fastify')().listen({host: undefined, port: 3000})
```
Which works, but produces a deprecation warning due to `dns.lookup()` not liking falsy `hostname` values (since node v12 - [code](https://github.com/nodejs/node/blob/cf2bce63862d995d238802ff4a7a24966361360a/lib/dns.js#L200:L201)):
```
> (node:156681) [DEP0118] DeprecationWarning: The provided hostname "undefined" is not a valid hostname, and is supported in the dns module solely for compatibility.
    at emitInvalidHostnameWarning (node:internal/dns/utils:270:13)
    at Object.lookup (node:dns:195:5)
    at Object.multipleBindings (/home/driimus/personal/fastify-test/node_modules/.pnpm/fastify@4.28.1/node_modules/fastify/lib/server.js:140:7)
    at /home/driimus/personal/fastify-test/node_modules/.pnpm/fastify@4.28.1/node_modules/fastify/lib/server.js:116:30
    at new Promise (<anonymous>)
    at /home/driimus/personal/fastify-test/node_modules/.pnpm/fastify@4.28.1/node_modules/fastify/lib/server.js:114:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

## Proposed change
I think it would be nice to get rid of the warning by re-assigning `listenOptions.host` even if the property already exists, in https://github.com/fastify/fastify/blob/9226a4793311c003e30dfddec4fd580deffcd28c/lib/server.js#L58-L60
As long as the provided value is nullish, which would more closely match the fallback for `host`:
https://github.com/fastify/fastify/blob/9226a4793311c003e30dfddec4fd580deffcd28c/lib/server.js#L54

## Use case
Looking back at the original example, a more likely use case is to override `host` using an env var or similar variable which is potentially undefined:
```js
 require('fastify')().listen({host: process.env.FASTIFY_HOST, port: 3000})
```

**Comment by metcoder95** [comment, marked as answer]
I can confirm the problem. Creating an issue from it

**Comment by driimus** [comment]
Closing this too to mirror #5583. Thanks all!
